crates/core/src/inline_snippets.rs

```diff
@@ -234,12 +234,28 @@ pub(crate) fn inline_sorted_snippets_with_offset(
         }
     }
 
+    let mut trailing_comma_lines = vec![];
```

Comment:
We already have a `delete_hanging_comma` function that should be used - we don't want two different pieces of code dealing with orphaned commas.

Any solution should either be [there](https://github.com/getgrit/gritql/blob/9587e9e5558c583d69514c1f581f8244ac551616/crates/core/src/inline_snippets.rs#L298-L374) or in a [replacement](https://github.com/getgrit/gritql/blob/9587e9e5558c583d69514c1f581f8244ac551616/crates/language/src/python.rs#L80).

Reply:
@morgante 
Had tried that earlier. I mean, placing the code fix in **delete_hanging_comma** function or in sub functions called inside it. That resulted in failure of an existing edge test case. This happens since **offsets** are again calculated after calling delete_hanging_comma and then actual replacements happens.

So, I feel doing this fix during actual replacements is the best approach.

Pasting here the change I was trying with :
just manipulated the range indexes here.
  
```
fn delete_hanging_comma(
    code: &str,
    replacements: &mut [(EffectRange, String)],
    offset: usize,
) -> Result<(String, Vec<ReplacementInfo>)> {
    // Handle the case when after applying replacements, a single comma
    // gets left behind on a single line
    let mut temp_code = code.to_string();
    for (range, snippet) in replacements.iter_mut() {
        let mut adjusted_range = adjust_range(&range.effective_range(), offset, &temp_code)?;
        if adjusted_range.start > temp_code.len() || adjusted_range.end > temp_code.len() {
            bail!("Range {:?} is out of bounds for code:\n{}\n", adjusted_range, temp_code);
        }

        temp_code.replace_range(adjusted_range.clone(), snippet);

        let line_start = temp_code[..adjusted_range.start].rfind('\n').map_or(0, |pos| pos + 1);
        let line_end = temp_code[adjusted_range.start..].find('\n').map_or(temp_code.len(), |pos| adjusted_range.start + pos);
        let line_content = temp_code[line_start..line_end].trim();
        
        // Check for lines with a single comma after replacements
        if line_content == "," {
            // Ensure the next character is a newline
            if line_end < temp_code.len() && temp_code.chars().nth(line_end) == Some('\n') {
                adjusted_range.end = line_end + 1;
            }

            // Update the original range to replace the range with an empty string later.
            range.range.end += adjusted_range.end - line_start;
        }
    }
```

Reply:
You should be able to do it in one of those two locations. Please don't make our handling even more spaghetti.

Reply:
@morgante  hey one doubt I have :

In below test case (existing), shouldnt we remove the empty newline?

```
#[test]
fn only_deduplicate_rewrites6() {
    run_test_expected({
        TestArgExpected {
            pattern: r#"
                engine marzano(0.1)
                language json
                pair($key, $value) => . where {
                    $key <: `"extends"`,
                }"#
            .to_owned(),
            source: r#"
            {
                "env": "bar",
                "extends": "foo",
                "baz": 1
            }"#
            .to_owned(),
            expected: r#"
            {
                "env": "bar",
                                                     <----------------------------------
                "baz": 1
            }"#
            .to_owned(),
        }
    })
    .unwrap();
}
```

Reply:
No, it is not necessary for Grit to handle removing newlines. We rely on formatters to handle that.

Reply:
@morgante got it. This fix should be satisfactory, have a look..